tests/NuGet.Services.AzureSearch.Tests/SearchService/SearchTextBuilderFacts.cs

```diff
@@ -102,6 +102,46 @@ public void ThrowsWhenTermIsTooBig(string query)
             }
         }
 
+        public class Autocomplete : FactsBase
+        {
+            // TODO: This should use the autocomplete package id field
+            // See https://github.com/NuGet/NuGetGallery/issues/6972
+            [Theory]
+            [InlineData("Test", "packageId:Test*")]
+            [InlineData("Test ", "packageId:Test*")]
+            [InlineData("title:test", "packageId:title\\:test*")]
+            [InlineData("Hello world", "packageId:Hello\\ world*")]
+            [InlineData("Hello world ", "packageId:Hello\\ world*")]
+            public void PackageIdAutocomplete(string input, string expected)
+            {
+                var request = new AutocompleteRequest
+                {
+                    Query = input,
+                    Type = AutocompleteRequestType.PackageIds
+                };
+
+                var result = _target.Autocomplete(request);
+
+                Assert.Equal(expected, result);
+            }
+
+            [Theory]
+            [InlineData("Test", "packageId:Test")]
+            [InlineData("Hello world", @"packageId:""Hello world""")]
+            public void PackageVersionsAutocomplete(string input, string expected)
+            {
+                var request = new AutocompleteRequest
+                {
+                    Query = input,
+                    Type = AutocompleteRequestType.PackageVersions
+                };
+
+                var result = _target.Autocomplete(request);
+
+                Assert.Equal(expected, result);
```

Comment:
The expected result does not have version. Should not be ?

Reply:
This tests the Azure Search query we generate, but doesn't test the result of running the Azure Search query. Please let me know if I misunderstood your question!